GitCommands/Git/GitModule.cs

```diff
@@ -2314,12 +2315,13 @@ public IReadOnlyList<GitStash> GetStashes(bool noLocks = false)
             return GetPatch(patches, fileName, oldFileName);
         }
 
-        public string GetRangeDiff(
+        public async Task<string> GetRangeDiffAsync(
             ObjectId firstId,
             ObjectId secondId,
             ObjectId? firstBase,
             ObjectId? secondBase,
-            string extraDiffArguments)
+            string extraDiffArguments,
+            CancellationToken cancellationToken)
```

Comment:
We should pass it into `_gitExecutable.ExecuteAsync`, and terminate the process if necessary. Is this something you planning to do separately?

Reply:
This is exactly the change that should be done with #9522, in the PR that is merged last.

I believe that the Execute() cancellation need to go together with a case where it is used and I implemented CustomMergeDiffTools first.
This change is simpler and could have be done first.